File: web-no-modal-sdk/modular-examples/auth0-react-modular-no-modal-example/app/src/config/chainConfig.ts
import { CHAIN_NAMESPACES, CustomChainConfig } from "@web3auth/base";

export const CHAIN_CONFIG = {
  mainnet: {
    chainNamespace: CHAIN_NAMESPACES.EIP155,
    chainId: "0x1", // Please use 0x1 for Mainnet
    rpcTarget: "https://rpc.ankr.com/eth",
    displayName: "Ethereum Mainnet",
    blockExplorer: "https://etherscan.io/",
    ticker: "ETH",
    tickerName: "Ethereum",
  } as CustomChainConfig,
  solana: {
    chainNamespace: CHAIN_NAMESPACES.SOLANA,
    chainId: "0x3", // Please use 0x1 for Mainnet, 0x2 for Testnet, 0x3 for Devnet
    rpcTarget: "https://api.devnet.solana.com",
    displayName: "Solana Devnet",
    blockExplorer: "https://explorer.solana.com",
    ticker: "SOL",
    tickerName: "Solana Token",
  } as CustomChainConfig,
  polygon: {
    chainNamespace: CHAIN_NAMESPACES.EIP155,
    rpcTarget: "https://polygon-rpc.com",
    blockExplorer: "https://polygonscan.com/",
    chainId: "0x89",
    displayName: "Polygon Mainnet",
    ticker: "matic",
    tickerName: "Matic",
  } as CustomChainConfig,
} as const;

export type CHAIN_CONFIG_TYPE = keyof typeof CHAIN_CONFIG;

File: flutter/flutter-playground/lib/features/home/domain/entities/chain_config.dart
import 'package:web3auth_flutter/enums.dart';

class ChainConfig {
  final ChainNamespace chainNamespace;
  final String displayName;
  final String ticker;
  final String rpcTarget;
  final String blockExplorerUrl;
  final String chainId;
  final String? logo;
  final bool isEVMChain;
  final String wss;

  ChainConfig({
    required this.isEVMChain,
    required this.chainNamespace,
    required this.displayName,
    required this.ticker,
    required this.rpcTarget,
    required this.blockExplorerUrl,
    required this.logo,
    required this.chainId,
    required this.wss,
  });
}
